volume-group-operator/controllers/volumegroup_controller.go

```diff
@@ -68,3 +86,53 @@ func (r *VolumeGroupReconciler) SetupWithManager(mgr ctrl.Manager, cfg *config.D
 		For(&volumegroupv1.VolumeGroup{}).
 		WithEventFilter(pred).Complete(r)
 }
+
+func (r *VolumeGroupReconciler) waitForCrds() error {
+	logger := volumeGroupLog.WithName("waitForCrds")
+
+	err := r.waitForVolumeGroupResource(logger, VolumeGroup)
+	if err != nil {
+		logger.Error(err, "failed to wait for VolumeGroup CRD")
+
+		return err
+	}
+
+	err = r.waitForVolumeGroupResource(logger, VolumeGroupClass)
+	if err != nil {
+		logger.Error(err, "failed to wait for VolumeGroupClass CRD")
+
+		return err
+	}
+
+	err = r.waitForVolumeGroupResource(logger, VolumeGroupContent)
+	if err != nil {
+		logger.Error(err, "failed to wait for VolumeGroupContent CRD")
+
+		return err
+	}
+
+	return nil
+}
+
+func (r *VolumeGroupReconciler) waitForVolumeGroupResource(logger logr.Logger, resourceName string) error {
+	unstructuredResource := &unstructured.UnstructuredList{}
+	unstructuredResource.SetGroupVersionKind(schema.GroupVersionKind{
+		Group:   volumegroupv1.GroupVersion.Group,
+		Kind:    resourceName,
+		Version: volumegroupv1.GroupVersion.Version,
+	})
+	for {
+		err := r.Client.List(context.TODO(), unstructuredResource)
+		if err == nil {
+			return nil
+		}
+		// return errors other than NoMatch
+		if !meta.IsNoMatchError(err) {
+			logger.Error(err, "got an unexpected error while waiting for resource", "Resource", resourceName)
+
+			return err
+		}
+		logger.Info("resource does not exist", "Resource", resourceName)
+		time.Sleep(5 * time.Second)
+	}
+}
```

Comment:
can you add a new line?
![image](https://user-images.githubusercontent.com/45543087/208376904-6609c932-b588-487c-b487-ed7680f8a20f.png)

Reply:
Done